packages/host/app/resources/room.ts

```diff
@@ -120,32 +104,63 @@ export class RoomResource extends Resource<Args> {
     }
   });
 
+  // note that the arrays below recreated as they are recomputed and hence
+  // different when the values change. downstream components that consume these
+  // messages directly as args are unnecessarily re-created since they are
+  // triple equals different when the values change. components should consume
+  // something stable like this resource and not these ever changing arrays
+  @cached
   get messages() {
-    return [...this._messageCache.values()].sort(
-      (a, b) => a.created.getTime() - b.created.getTime(),
-    );
+    if (this.roomId == undefined) {
+      return [];
+    }
+    return [...this._messageCache.values()]
+      .filter((m) => m.roomId === this.roomId)
+      .sort((a, b) => a.created.getTime() - b.created.getTime());
   }
 
+  @cached
   get members() {
-    return Array.from(this._memberCache.values()) ?? [];
+    if (this.roomId == undefined) {
+      return [];
+    }
+    return (
+      Array.from(this._memberCache.values()).filter(
+        (m) => m.roomId === this.roomId,
+      ) ?? []
+    );
   }
 
+  @cached
   get invitedMembers() {
-    return this.members.filter((m) => m.membership === 'invite');
+    if (this.roomId == undefined) {
+      return [];
+    }
+    return this.members.filter(
+      (m) => m.membership === 'invite' && m.roomId === this.roomId,
+    );
   }
 
+  @cached
   get joinedMembers() {
-    return this.members.filter((m) => m.membership === 'join');
+    if (this.roomId == undefined) {
+      return [];
+    }
+    return this.members.filter(
+      (m) => m.membership === 'join' && m.roomId === this.roomId,
+    );
```

Comment:
note that these caches span rooms. that is fine, it makes for quick access when switching between rooms, but it means that we should filter these values by room and use a `@cached` decorator so we only do this filtering when the room switches--not for every access of the property